cosmos/models/Workflow.py

```diff
@@ -560,62 +587,43 @@ def _run(workflow, session, task_queue):
     Do the workflow!
     """
     workflow.log.info('Executing TaskGraph')
-
-    watcher = SignalWatcher(workflow)
-
-    # graph_failed = nx.DiGraph()
-    #
-    # def handler(signal, frame):
-    #     task_queue.add_edges(graph_failed.edges())
-    #     for task in graph_failed.nodes():
-    #         task.attempt +=1
-    #         task.status = TaskStatus.no_attempt
-    #     graph_failed.remove_nodes_from(graph_failed.nodes())
-
-    # signal.signal(signal.SIGUSR1, handler)
-
     available_cores = True
-    while len(task_queue) > 0:
-        if available_cores:
-            _run_queued_and_ready_tasks(task_queue, workflow)
-            available_cores = False
-
-        for task in _process_finished_tasks(workflow.jobmanager):
-            if task.status == TaskStatus.failed and task.must_succeed:
-
-                if workflow.info['fail_fast']:
-                    workflow.log.info('Exiting run loop at first Task failure')
-                    workflow.terminate(due_to_failure=True)
-                    return
-
-                # pop all descendents when a task fails; the rest of the graph can still execute
-                remove_nodes = descendants(task_queue, task).union({task, })
-                # graph_failed.add_edges(task_queue.subgraph(remove_nodes).edges())
-
-                task_queue.remove_nodes_from(remove_nodes)
-                workflow.status = WorkflowStatus.failed_but_running
-                workflow.log.info('%s tasks left in the queue' % len(task_queue))
-            elif task.status == TaskStatus.successful:
-                # just pop this task
-                task_queue.remove_node(task)
-            elif task.status == TaskStatus.no_attempt:
-                # the task must have failed, and is being reattempted
-                pass
-            else:
-                raise AssertionError('Unexpected finished task status %s for %s' % (task.status, task))
-            available_cores = True
-
-        # only commit Task changes after processing a batch of finished ones
-        session.commit()
-        watcher.wait(.3)
-        if watcher.caught_signal():
-            workflow.log.info('Interrupting workflow to handle signal %d', watcher.last_signal)
-            workflow.terminate(due_to_failure=False)
-            return
-
-
-import networkx as nx
 
```

Comment:
This `while len(task_queue) > 0` block is 100% red/green because I nested it in a with-statement. The only changes are at the very beginning and at the very end.